### PR TITLE
Fix conda env, remove static binary dependency where possible, add documentation for installation

### LINKED ISSUE #5: Solving environment: failed
Why did I get this warning?

(CRISPRtracrRNA) PS C:\Users\oz\CRISPRtracrRNA> conda env create -f environment.yml
Collecting package metadata (repodata.json): done
Solving environment: failed

ResolvePackageNotFound:
  - libpng==1.6.37=hed695b0_2
  - py-boost==1.73.0=py37ha9443f7_11
  - scipy==1.4.1=py37ha3d9a3c_3
  - perl-threaded==5.26.0=0
  - xorg-libxau==1.0.9=h14c3975_0
  -

### PR DESCRIPTION
**Pull request by richardstoeckl**
Hi Alex,

This PR does three interconnected things:

## 1. removes the static binaries for `cmscan` and `fasta36` and moves these dependencies to the conda environment for ease of use.

Some users reported errors regarding the permissions of these binaries (see issues #3 , #7 ). Switching these to the conda environment solves this and makes handling the dependencies easier.

## 2. fixes the conda environment for linux users (tested with Ubuntu 22.04.5 LTS).

This should also fix #5 

Please note:
This environment will not work with `osx-64` systems, as the `anaconda::tensorflow` repo only has support for `osx-64`since version ~`2.9.0` and the `conda-forge::tensorflow` repo does not contain the version `2.3` at all. Maybe upgrading to `conda-forge::tensorflow==2.4.0` could be an option in the future for further compatibility.

## 3. provides a bit more documentation on how to avoid errors during installation

- Some download methods result in the `erpin` binary having the wrong permissions set up. Unfortunately, this software is not maintained and not available via conda, so I could not move it to the environment like `cmscan` and `fasta36`. I added instructions to the README on how to give the required permissions.
- Sometimes, the cas analysis via `CRISPRcasIdentifier` fails. A workaround is to manually extract the HMM and ML models tar.gz files. I added a note for this in the README.

Best,
Richard

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -1,155 +1,51 @@
...
 channels:
-  - biobuilds
-  - bioconda
-  - anaconda
   - conda-forge
-  - defaults
+  - bioconda
+  - nodefaults
+  - biobuilds
+  - r
```
Comment (bgruening) [inline environment.yml]: Do we really need those two?

File: environment.yml [hunk 2]
```diff
@@ -1,155 +1,51 @@
...
-  - xgboost=1.2.0
-  - xorg-libxau=1.0.9
-  - xorg-libxdmcp=1.1.3
-  - xz=5.2.5
-  - yaml=0.2.5
-  - yarl=1.6.2
-  - zipp=3.4.0
-  - zlib=1.2.11
-  - zstd=1.4.9
+  - anaconda::tensorflow==2.3.0
```
Comment (bgruening) [inline environment.yml:36]: this should come from conda-forge

**Review by bgruening — COMMENTED** [review]
Very nice! Thanks! Much needed.

File: environment.yml (hunk 2 above)
Comment (richardstoeckl) [reply environment.yml:36]: You are right. However, as written in my PR:

>  the anaconda::tensorflow repo only has support for osx-64since version ~2.9.0 and the conda-forge::tensorflow repo does not contain the version 2.3 at all. Maybe upgrading to conda-forge::tensorflow==2.4.0 could be an option in the future for further compatibility.

Unfortunately, tensorflow 2.3.0 is not available with conda-forge. Thats why I had to use this way.

File: environment.yml (hunk 1 above)
Comment (richardstoeckl) [reply environment.yml]: Good point! I will test the env without them.

File: environment.yml (hunk 2 above)
Comment (bgruening) [reply environment.yml:36]: How complicated is it to update the code?

The problem is that you remove the default channel, which is good! But then you take this dependency out of the default channel.

Comment (richardstoeckl) [reply environment.yml:36]: I don't know if changing the tensorflow version has any effects on the results that are not obvious through patch notes. The [release notes](https://github.com/tensorflow/tensorflow/releases/tag/v2.4.0) mention plenty of breaking changes though. Several big parts of this tool (including CRISPRidentify and CRISPRcasIdentifier) depend on tensorflow/keras, and I don't have extensive test data and results that would be required to check if any edge cases are affected.

I would argue that updating and testing all of these tools to a newer version of tensorflow is out of scope of this PR. Leaving this line as it is just makes sure it works now, and leaves the possibility to easily remove the default channel in the future.

File: environment.yml (hunk 1 above)
Comment (richardstoeckl) [reply environment.yml]: Since I had to make sure CRISPRidentify works within this env as well, I based the dependencies on that. The channels were added in [this commit to CRISPRidentify](https://github.com/BackofenLab/CRISPRidentify/commit/6fa5de3ba1bede34d4a282e3f6df97c69af7f252), but as you suspected, they don't seem to be required. The environment resolved without them and the results stayed the same.

**Review by Alexander-Mitrofanov — APPROVED** [review]
Great job!
